src/App.tsx

```diff
@@ -26,27 +24,78 @@ interface Trail {
   filename: string;
 }
 
+interface QueryParams {
+  [key: string]: string | number | undefined;
+}
+
 const getTrailData = (trailFilename: string) => 
   fetch(`data/trails/${trailFilename}`)
   .then((res) => res.json())
   .then((res) => res.features.find((feature: Feature) => feature.geometry.type === "MultiLineString"))
 ;
 
+const getSearchResult = (search: string)=> {
+  const baseURL = new URL(`https://api.mapbox.com/geocoding/v5/mapbox.places/${search}.json`)
+  const queryParams: QueryParams = {
+    "access_token": process.env.REACT_APP_MAPBOX_TOKEN,
+    "limit": 10
+  }
+  Object.keys(queryParams).forEach((key) => 
+    baseURL.searchParams.append(key, queryParams[key] as string)
+  )
+  return fetch(baseURL as unknown as string).then((response) => response.json())
+};
+
 function App() {
   const [selectedTrail, setSelectedTrail] = useState<Feature<MultiLineString> | undefined>(undefined);
+  const [autoCompleteList, setAutoCompleteList] = useState<FeatureCollection | undefined>(undefined);
+  const [searchResult, setSearchResult] = useState<FeatureCollection | undefined>(undefined);
```

Comment:
I don't think you need to explicitly pass `undefined` when initializing state. Leaving it empty should set it as `undefined`
```suggestion
  const [selectedTrail, setSelectedTrail] = useState<Feature<MultiLineString> | undefined>();
  const [autoCompleteList, setAutoCompleteList] = useState<FeatureCollection | undefined>();
  const [searchResult, setSearchResult] = useState<FeatureCollection | undefined>();
```